test/e2e/windows/framework.go

```diff
@@ -17,19 +17,26 @@ limitations under the License.
 package windows
 
 import (
+	"k8s.io/kubernetes/test/e2e/framework"
 	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
 
 	"github.com/onsi/ginkgo/v2"
 )
 
 // SIGDescribe annotates the test with the SIG label.
-func SIGDescribe(text string, body func()) bool {
-	return ginkgo.Describe("[sig-windows] "+text, func() {
+func SIGDescribe(text string, args ...interface{}) bool {
+	if len(args) == 0 {
+		panic("need additional parameters")
+	}
+	// Callback must come last.
+	body := args[len(args)-1].(func())
+	args[len(args)-1] = func() {
```

Comment:
is this convention guaranteed? this looks pretty complex to understand, lot of magic here

Reply:
why is different though? because of the BeforeEach?

I think that the comment should assume readers don't have any idea of how ginkgo works,

Reply:
> Is this convention guaranteed?

It's true for all callers of this particular `SIGDescribe` and also used everywhere else.

> why is different though? because of the BeforeEach?

Yes. That has to be added before the original callback.

Reply:
I got dissatisfied with the amount of magic required here. I've added a second commit with a simpler solution (wrap the callback to inject the `BeforeEach`), albeit that now touches more files.

@aojea: I'll let you decide what you prefer.

Reply:
I found this last iteration simpler and less invasive